src/gui/tray/UserLine.qml

```diff
@@ -173,13 +175,19 @@ MenuItem {
                             accountMenu.close()
                         }
 
+                        contentItem: Label {

```

Comment:
@FlexW Now, as we are adding more and more duplicate code into every **MenuItem** I am starting to think that we need to factor it out of **Menu**, inherit our custom **MenuItem** from the former, and use either of
- `Menu::contentItem` https://doc.qt.io/qt-5/qml-qtquick-controls2-menu.html#contentModel-prop
- **Repeater** https://doc.qt.io/qt-5/qml-qtquick-repeater.html
- **Instantiator** https://doc.qt.io/archives/qt-5.9/qml-qtqml-instantiator.html

Not sure which one suits best. Probably model. I guess I did the same for the "three-dot-menu" on activity list items some months ago :)

So, the idea would be to fill the **Menu** with **MenuItems** without duplicating the entire **MenuItem** child for each menu item. Instead, using a template to create a **MenuItem** with only a different name and **onPressed** callback. The latter can also be implemented via a single **Q_INVOKABLE** that accepts an index parameter (or a string with the short name of a **MenuItem**, whatever) and it can get processed on the C++ side. I did the same for activity list items.

NOTE: Not necessarily doing this in the current PR. Even better to implement it later in a separate one.

Reply:
Yes, this is long overdue in here. That QML code needs a big clean up. Too many things are too ad hoc. You can also add to the list: less Q_INVOKABLE and more Q_PROPERTY for the integration with the C++.

Reply:
Ah also: this depends on QQC2 without using at all its styling abilities. Instead of spreading styling code everywhere use QQC2 properly? Or just don't use it.

Reply:
@allexzander that makes sense. However, this can quickly go out of scope of the pr. Should we first do a pr that cleans that up or should I add that in a different commit?